editor/animation_track_editor_plugins.cpp

```diff
@@ -1086,8 +1087,24 @@ void AnimationTrackEditTypeAudio::gui_input(const Ref<InputEvent> &p_event) {
 	}
 
 	if (len_resizing && mm.is_valid()) {
+		// Rezising index is some
 		len_resizing_rel += mm->get_relative().x;
-		len_resizing_start = mm->is_shift_pressed();
+		float ofs_local = len_resizing_rel / get_timeline()->get_zoom_scale();
+		float prev_ofs_start = get_animation()->audio_track_get_key_start_offset(get_track(), len_resizing_index);
+		float prev_ofs_end = get_animation()->audio_track_get_key_end_offset(get_track(), len_resizing_index);
+		Ref<AudioStream> stream = get_animation()->audio_track_get_key_stream(get_track(), len_resizing_index);
+		float len = stream->get_length();
+		if (len == 0) {
+			Ref<AudioStreamPreview> preview = AudioStreamPreviewGenerator::get_singleton()->generate_preview(stream);
+			float preview_len = preview->get_length();
+			len = preview_len;
+		}
+
+		if (len_resizing_start)
+			len_resizing_rel = CLAMP(ofs_local, -prev_ofs_start + 0.01, len - prev_ofs_end - prev_ofs_start - 0.01) * get_timeline()->get_zoom_scale();
+		else
+			len_resizing_rel = CLAMP(ofs_local, -(len - prev_ofs_end - prev_ofs_start) + 0.01, prev_ofs_end - 0.01) * get_timeline()->get_zoom_scale();
```

Comment:
Our style guide requires brackets even for single-line `if`/`else` blocks.

Reply:
https://github.com/godotengine/godot/pull/69302/commits/70abca3a9e2fc3a8926a1bfec5951b2e22d407fa